flink-connector-aws/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/fanout/FanOutShardSubscriber.java

```diff
@@ -249,22 +262,21 @@ private FanOutShardSubscription openSubscriptionToShard(final StartingPosition s
 
         kinesis.subscribeToShard(request, responseHandler);
 
-        boolean subscriptionEstablished =
-                waitForSubscriptionLatch.await(
+        boolean subscriptionTimedOut =
+                !waitForSubscriptionLatch.await(
                         subscribeToShardTimeout.toMillis(), TimeUnit.MILLISECONDS);
 
-        if (!subscriptionEstablished) {
+        if (subscriptionTimedOut) {
             final String errorMessage =
                     "Timed out acquiring subscription - " + shardId + " (" + consumerArn + ")";
             LOG.error(errorMessage);
             subscription.cancelSubscription();
-            handleError(
-                    new RecoverableFanOutSubscriberException(new TimeoutException(errorMessage)));
```

Comment:
I believe the intention here might have been to retry this indefinitely. But I believe by wrapping it in RecoverableFanOutSubscriberException, we might have made it retriable (retry count subject to provided retry config)

Reply:
By wrapping this makes it retry forever

```
// Recoverable errors should be reattempted without contributing to the retry policy
// A recoverable error would not result in the Flink job being cancelled
```

https://github.com/apache/flink-connector-aws/blob/main/flink-connector-aws/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/fanout/FanOutRecordPublisher.java#L185

Reply:
Ah I see the bug, nice find. It is being wrapped again in `handleError`